gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+            "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName));
+      }
+      // zone exists but change does not
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.changeId' resource named '%s' does not exist.", changeId));
+    }
+    Change result = OptionParsersAndExtractors.extractFields(change, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(String.format(
+          "Error when serializing change %s in managed zone %s in project %s.",
+          changeId, zoneName, projectId));
+    }
+  }
+
+  /**
+   * Returns a response to getZone service call.
+   */
+  Response getZone(String projectId, String zoneName, String[] fields) {
+    ZoneContainer container = findZone(projectId, zoneName);
+    if (container == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName));
+    }
+    ManagedZone result = OptionParsersAndExtractors.extractFields(container.zone(), fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(String.format(
+          "Error when serializing managed zone %s in project %s.", zoneName, projectId));
+    }
+  }
+
+  /**
+   * We assume that every project exists. If we do not have it in the collection yet, we just create
+   * a new default project instance with default quota.
+   */
+  Response getProject(String projectId, String[] fields) {
+    ProjectContainer defaultProject = createProject(projectId);
+    projects.putIfAbsent(projectId, defaultProject);
+    Project project = projects.get(projectId).project(); // project is now guaranteed to exist
+    Project result = OptionParsersAndExtractors.extractFields(project, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing project %s.", projectId));
+    }
+  }
+
+  /**
+   * Creates a project. It generates a project number randomly (we do not have project numbers
+   * available).
+   */
+  private ProjectContainer createProject(String projectId) {
+    Quota quota = new Quota();
+    quota.setManagedZones(10000);
+    quota.setRrsetsPerManagedZone(10000);
+    quota.setRrsetAdditionsPerChange(100);
+    quota.setRrsetDeletionsPerChange(100);
+    quota.setTotalRrdataSizePerChange(10000);
+    quota.setResourceRecordsPerRrset(100);
+    Project project = new Project();
+    project.setId(projectId);
+    project.setNumber(new BigInteger(String.valueOf(
+        Math.abs(ID_GENERATOR.nextLong() % Long.MAX_VALUE))));
+    project.setQuota(quota);
+    return new ProjectContainer(project);
+  }
+
+  /**
+   * Deletes a zone.
```

Comment:
In general, unless the javadoc provides some non-obvious details about the method, I think we can remove javadoc for private and package scope methods (like here).